test/sim/moves/perishsong.js

```diff
@@ -38,4 +38,26 @@ describe('Perish Song', function () {
 		for (let i = 0; i < 4; i++) { battle.makeChoices(); }
 		assert.equal(battle.winner, 'Player 2');
 	});
+
+	it(`should not affect other Pokemon with the ability Soundproof`, function () {
+		battle = common.createBattle([[
+			{species: 'Weavile', ability: 'soundproof', moves: ['sleeptalk']},
+		], [
+			{species: 'Slowpoke', ability: 'soundproof', moves: ['perishsong']},
+		]]);
+
+		for (let i = 0; i < 4; i++) { battle.makeChoices(); }
+		assert.equal(battle.winner, 'Player 1');
```

Comment:
I think it would be much more straightforward to just check the if the Pokemon has the Perish Song volatile. That way, the test is independent of Perish Song's fainting behavior. You can do ``.volatiles['perishsong']`` to check if it currently has Perish Song (see Protective Pads test vs. Perish Body as an example).

Reply:
I was just copying the other test because I suck at writing tests... I'll see what I can do.

Reply:
Well, I did it, but I guess I should have copied and pastd  `battle.p1.active[0].volatiles['perishsong]'` to save me all of that typing...